pandas/core/generic.py

```diff
@@ -6885,14 +6885,48 @@ def fillna(
                         if not is_dict
                         else downcast.get(k)  # type: ignore[union-attr]
                     )
-                    # GH47649
-                    result.loc[:, k] = (
-                        result[k].fillna(v, limit=limit, downcast=downcast_k).values
-                    )
-                    # TODO: result.loc[:, k] = result.loc[:, k].fillna(
-                    #     v, limit=limit, downcast=downcast_k
-                    # )
-                    # Revert when GH45751 is fixed
```

Comment:
Is #45751 still applicable to this code path/is it fixed?

Reply:
I don't think so, no